### PR TITLE
Game

### PR DESCRIPTION
**Pull request by Acusanno**
Updated the back-end:
- The game now has rooms
- The players need to be identified and to press add to queue in the game page
- Once two players are added to the queue, the game is instantiated and both players are redirected to Game/roomId
- Games now end when a player has 3 points, later you should choose maxScore before making a custom game
- When the game end, the result object is created (but not sent anywhere yet) and the back close the room instance

### CODE REVIEW

File: back/src/game/pong.gateway.ts [hunk 1]
```diff
@@ -12,29 +13,53 @@ export interface Move {
...
+		console.log(this.gameService.gameRoom.length);
+		if (this.gameService.gameRoom.length > 0)
+			this.gameService.EndGame(client.id);
 	}
 
 	@SubscribeMessage("Move1")
 	HandleMove1(@MessageBody() input: Move, @ConnectedSocket() client: Socket)
 	{
-		this.gameService.updateMove1(input);
+		if (this.gameService.gameRoom.length > 0)
```
Comment (se-lee) [inline back/src/game/pong.gateway.ts:55]: delete this condition (not needed)

File: back/src/game/pong.gateway.ts [hunk 2]
```diff
@@ -12,29 +13,53 @@ export interface Move {
...
+			this.gameService.updateMove1(input, client.id);
 	}
 
 	@SubscribeMessage("Move2")
 	HandleMove2(@MessageBody() input: Move, @ConnectedSocket() client: Socket)
 	{
-		this.gameService.updateMove2(input);
+		if (this.gameService.gameRoom.length > 0)
```
Comment (se-lee) [inline back/src/game/pong.gateway.ts:62]: delete this condition (not needed)

**Review by se-lee — APPROVED** [review]
game room working ok, be sure to remove unnecessary lines of code (conditions, comment outs unless you really want to keep them)

**Review by Blodispic — APPROVED** [review]
Beaucoup de fonctions passé en commentaire, mais sinon tout me parait bon.

**Pull request merged**